http-server/src/main/java/org/triplea/server/lobby/game/listing/GameListing.java

```diff
@@ -163,42 +98,35 @@ private void removeGame(final GameId gameId) {
    *     their game. Otherwise true indicates the game is present and the keep-alive period has been
    *     extended.
    */
-  boolean keepAlive(final String apiKey, final String gameId) {
-    final var gameEntry =
-        games.entrySet().stream().filter(entry -> entry.getKey().id.equals(gameId)).findAny();
-    if (gameEntry.isEmpty()) {
-      log.warn("Keep alive received for DEAD game: {}", gameId);
+  boolean keepAlive(final ApiKey apiKey, final String gameId) {
+    final Optional<LobbyGame> game =
+        Optional.ofNullable(games.getIfPresent(new GameId(apiKey, gameId)));
+    if (game.isEmpty()) {
+      log.warn("Keep alive for removed game: {}", gameId);
       return false;
     }
-
-    final var id = gameEntry.get().getKey();
-
-    if (!id.apiKey.equals(apiKey)) {
-      throw new IncorrectApiKey();
-    }
-    gameReaper.registerKeepAlive(gameId);
-    log.info("Keep alive received for game: {}", gameId);
+    updateGame(apiKey, gameId, game.get());
     return true;
   }
 
   /** Moderator action to remove a game. */
   void bootGame(final int moderatorId, final String gameId) {
-    final GameId gameToRemove =
-        games.entrySet().stream()
-            .filter(entry -> entry.getKey().id.equals(gameId))
-            .findAny()
-            .map(Map.Entry::getKey)
-            .orElseThrow(() -> new GameNotFound(gameId));
-
-    final String hostName = games.get(gameToRemove).getHostName();
-
-    removeGame(gameToRemove);
-    log.info("Moderator {} booted game: {}, hosted by: {}", moderatorId, gameId, hostName);
-    auditHistoryDao.addAuditRecord(
-        ModeratorAuditHistoryDao.AuditArgs.builder()
-            .moderatorUserId(moderatorId)
-            .actionName(ModeratorAuditHistoryDao.AuditAction.BOOT_GAME)
-            .actionTarget(hostName)
-            .build());
+    games.asMap().entrySet().stream()
+        .filter(entry -> entry.getKey().id.equals(gameId))
```

Comment:
This really makes me feel like there should be 2 different maps
1. gameId string -> API key
2. gameId string -> LobbyGame

Instead of
gameId object -> Lobby game

Reply:
The fact that 'post' game generates a game id prevents us from having duplicates, but in theory it would be okay for there to be a duplicate gameId so long as the API keys were different. On the client side that would not work, hence it's not allowed. More fundamentally though the 'game id' is a true compound key, searching by gameId and wrong API key returns no results. 

The double map I see as having a few problems:
- multiple caches that need to be kept in sync
- parallel data structure problem (it's similar to two arrays where there is correlation between the pair-wise elements)
- logic for not found when gameId is there but API key is not would be a bit more complicated. Recall that gameId is publicly known, someone could try to update a game by passing a gameId, but their API key would not match. The lookup logic would be a bit more complicated rather than a simple 'not found' case.